ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotSelectorFromPeers.java

```diff
@@ -86,41 +87,24 @@ protected Optional<FastSyncState> fromBestPeer(final EthPeer peer) {
   }
 
   protected Optional<EthPeer> selectBestPeer() {
-    return ethContext
-        .getEthPeers()
-        .bestPeerMatchingCriteria(this::canPeerDeterminePivotBlock)
-        // Only select a pivot block number when we have a minimum number of height estimates
-        .filter(unused -> enoughFastSyncPeersArePresent());
-  }
-
-  private boolean enoughFastSyncPeersArePresent() {
-    final long peerCount = countPeersThatCanDeterminePivotBlock();
+    List<EthPeer> peers =
+        ethContext
+            .getEthPeers()
+            .streamAvailablePeers()
+            .filter((peer) -> peer.chainState().hasEstimatedHeight() && peer.isFullyValidated())
+            .toList();
+
+    // Only select a pivot block number when we have a minimum number of height estimates
     final int minPeerCount = syncConfig.getSyncMinimumPeerCount();
-    if (peerCount < minPeerCount) {
+    if (peers.size() < minPeerCount) {
       LOG.info(
           "Waiting for valid peers with chain height information.  {} / {} required peers currently available.",
-          peerCount,
+          peers.size(),
           minPeerCount);
-      return false;
+      return Optional.empty();
+    } else {
+      return peers.stream().max(ethContext.getEthPeers().getBestPeerComparator());
     }
-    return true;
-  }
-
-  private long countPeersThatCanDeterminePivotBlock() {
-    return ethContext
-        .getEthPeers()
-        .streamAvailablePeers()
-        .filter(this::canPeerDeterminePivotBlock)
-        .count();
-  }
-
-  private boolean canPeerDeterminePivotBlock(final EthPeer peer) {
-    LOG.debug(
```

Comment:
we're losing this log line, is that intentional?

Reply:
It's not a particularly good log as it doesn't include any context and is likely to end up listing peers. I can look to add something similar if you think it's worthwhile

Reply:
all good just checking it won't be missed